backend/src/db/models/user.ts

```diff
@@ -84,6 +84,31 @@ export async function getUser(userId: string): Promise<APIv4.ServerUser> {
     return user;
 }
 
+export async function updateUser(
+    userId: APIv4.UserId,
+    updateFields: Record<string, any>,
+): Promise<void> {
+    const lookup = await collections.users.findOne({ _id: userId });
+
+    if (lookup === null) {
+        throw new Error(`User with ID ${userId} not found`);
+    }
+
+    if (Object.keys(updateFields).length > 0) {
+        await collections.users.updateOne(
+            { _id: userId },
+            { $set: updateFields },
```

Comment:
We probably should sanitize the data at least a little bit before throwing it into the database. Otherwise we might accidentally have a data corruption down the line.

For example, if `updateFields` happen to have a method defined as part of the object (perfectly fine for a `Record<string, any>`), then this will attempt to write a function object into the database, which is obviously bad.

Reply:
I update the argument to be `Partial<APIv4.ServerUser>` which should only allow valid user partial to be taken as the `updateFields`. Do you see any some uncommon object type that could slip through this TS argument safeguard?